### PR TITLE
basic google authentication added

### PR DESCRIPTION
**Pull request by gdm-1718-jerobern**
Google authentication & Facebook authentication fetching first name and profile picture of user.
Minimal styling, no error handling.

### CODE REVIEW

**Comment by thibmaek** [comment]
Hi @gdm-1718-jerobern !

I quickly glanced through the PR and it looks good! There are some light remarks which I'll post later today but for now, thanks for adding this to the repo 🙏 

PS: you can ignore the failing build, this is due to something else.

File: App.js [hunk 1]
```diff
@@ -57,8 +57,15 @@ export default class App extends Component {
     });
   };
 
-  handleUserLogin = async () => {
-    const userInfo = await handleUserLogin();
+  handleFacebookLogin = async () => {
+    const userInfo = await handleFacebookLogin();
+    userInfo.picture = userInfo.picture.data.url;
```
Comment (thibmaek) [inline App.js]: This is clever! Indeed a nice way to get the same properties across the Facebook & Google logins.
But in general, we avoid directly reassigning the parameters of an object property as much as possible.

Instead you could leverage the Object spread syntax:

```js
this.setState({
	userInfo: {
		...userInfo,
		picture: userInfo.picture.data.url,
	}
})
```

Doing this you still keep the state as it returns from Facebook in our function but return our own preferred object to the state.

File: App.js [hunk 2]
```diff
@@ -57,8 +57,15 @@ export default class App extends Component {
...
+    this.setState({ userInfo });
+  };
+
+  handleGoogleLogin = async () => {
+    const userInfo = await handleGoogleLogin();
+    userInfo.first_name = userInfo.given_name;
```
Comment (thibmaek) [inline App.js]: Same here as the above comment ;)

File: utils/authenticationService.js [hunk 3]
```diff
@@ -1,14 +1,34 @@
...
+export const handleFacebookLogin = async () => {
   const { type, token } = await Expo.Facebook.logInWithReadPermissionsAsync(FB_APP_ID, {
     permissions: ['public_profile'],
     behavior: 'web',
   });
 
   if (type === 'success') {
     // Get the user's name using Facebook's Graph API
-    const userInfoResponse = await fetch(`https://graph.facebook.com/me?access_token=${token}&fields=id,name,picture.type(large)`);
+    const userInfoResponse = await fetch(`https://graph.facebook.com/me?access_token=${token}&fields=id,first_name,picture.type(large)`);
```
Comment (thibmaek) [inline utils/authenticationService.js]: Nice catch!

**Comment by gdm-1718-jerobern** [comment]
@thibmaek how can I "uncheck package-lock.json from this PR" ?

**Comment by thibmaek** [comment]
Just checked out this PR and tested it locally, looking gooood 👯 

image

**Comment by thibmaek** [comment]
@gdm-1718-jerobern no worries about the package-lock. I saw you deleted it, but you just had to undo the changes made. I can take that from here if you want :)

Sit back & relax if you want :)

**Pull request merged**